src/plugins/kubeconfig/types.ts

```diff
@@ -0,0 +1,74 @@
+/** Copyright © 2024-present P0 Security
+
+This file is part of @p0security/cli
+
+@p0security/cli is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 3 of the License.
+
+@p0security/cli is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with @p0security/cli. If not, see <https://www.gnu.org/licenses/>.
+**/
+import { PermissionSpec } from "../../types/request";
+
+export type K8sConfig = {
+  workflows: {
+    items: K8sClusterConfig[];
+  };
+};
+
+export type BaseK8sClusterConfig = {
+  clusterId: string;
+
+  clusterServer: string;
+  clusterCertificate: string;
+  state: string;
+} & (KubernetesProxyComponentConfig | KubernetesPublicComponentConfig);
+
+// k8s clusters that are not EKS clusters will not have these set...
+export type K8sClusterConfig = BaseK8sClusterConfig & {
```

Comment:
Hmm, is this actually a problem though? `EksClusterConfig` is just a more restrictive version of `K8sClusterConfig` (where `awsAccountId` and `awsClusterArn` are not allowed to be undefined); I don't think it's an issue to, for example, allow a `EksClusterConfig` object to be provided where a `K8sClusterConfig` is expected, and therefore I see no reason to not allow them to be used interchangeably.

Reply:
Though, to be clear, I'm not sure my approach here is the best one. I know I could have done

```typescript
export type EksClusterConfig = {
  [K in keyof K8sClusterConfig]-?: K8sClusterConfig[K];
} 
```

But I think this might be overkill, since there might be future fields added to `K8sClusterConfig` that `EksClusterConfig` need not care about.

Reply:
> Though, to be clear, I'm not sure my approach here is the best one. I know I could have done
> 
> ```ts
> export type EksClusterConfig = {
>   [K in keyof K8sClusterConfig]-?: K8sClusterConfig[K];
> } 
> ```
> 
> But I think this might be overkill, since there might be future fields added to `K8sClusterConfig` that `EksClusterConfig` need not care about.

You can use the built in typescript `Required<T>` to make optional parameters non-optional or `Partial<T>` to make no-optional parameters required and it achieves the same.